lighthouse-core/gather/gatherers/styles.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Gathers the active style and stylesheets used on a page.
+ * "Active" means that if the stylesheet is removed at a later time
+ * (before endStylesCollect is called), this gatherer will not include it.
+ */
+
+'use strict';
+
+const WebInspector = require('../../lib/web-inspector');
+const Gatherer = require('./gatherer');
+
+/**
+ * @param {!gonzales.AST} parseTree
+ * @return {!Array}
+ */
+function getCSSPropsInStyleSheet(parseTree) {
+  let results = [];
+
+  parseTree.traverseByType('declaration', function(node, index, parent) {
+    const [name, val] = node.toString().split(':').map(item => item.trim());
```

Comment:
We've avoided destructuring up till now. AFAIK it's not supported back to node 4 --harmony. Can we do this without destructuring?